vars/uploadArtifacts.groovy

```diff
@@ -2,6 +2,10 @@ void call(Map args = [:]) {
     def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
 
     def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: args.manifest))
+    def yamlFile = readYaml(file: args.manifest)
+    def fileLocation = yamlFile.components.artifacts.dist[0]
+    def productName = buildManifest.build.getFilename()
+    def fileName = buildManifest.getPackageName("${productName}")
```

Comment:
This seems to load the same YAML file a second time, then combine its properties. Extend `BuildManifest` to know about `components.artifacts.dist`, and implement it as a single method of `buildManifest`.